Services/UI/OJS.Services.Ui.Business/Implementations/UsersBusinessService.cs

```diff
@@ -63,6 +63,34 @@ public class UsersBusinessService : IUsersBusinessService
             return profile;
         }
 
+        public async Task<string?> GetUserIdByUsername(string? username)
```

Comment:
IMO this method is not needed at all. You already do the check for admin or lecturer before calling it.

Reply:
This method returns the user id for the username and it is needed to get the submissions in the logic below. Seems like i can remove the check for admin above, since it is contained in the `GetUserIdByUsername` method.